src/lib/command.rs

```diff
@@ -197,6 +198,8 @@ pub(crate) fn run_command(command_string: &str, args: &Option<Vec<String>>, vali
 pub(crate) fn run(step: &Step) {
     let validate = !step.config.should_ignore_errors();
 
+    recursion_level::increase_level();
```

Comment:
you are increasing it when invoking a task, not when invoking a child cargo-make.
so i'm not 100% sure this is the right way.
there are several places where cargo-make calls itself

1. watch task
2. protected flow
3. fork
(1-3 go through create_proxy_task)
4. workspace - so cargo-make creates a script that calls each member.
5. user created tasks - that can be in the command or script.

so as i see it, the best way to ensure this works bulletproof for all scenarios is to increment it in the cli.rs run function right from the beggining. and if not defined, set to 0.
wdyt?

Reply:
should be fixed now.